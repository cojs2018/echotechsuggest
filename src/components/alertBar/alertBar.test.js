import React from 'react';
import render from 'react-test-renderer';
import { mount } from 'enzyme';

import AlertBar from './alertBar';

describe('AlertBar', () => {

    it('renders correctly', () => {
        const props = {
            visible: true,
            setVisible: jest.fn((bool) => visible = bool),
            severity: 'success',
            message: 'This is a test alert message'
        }

        const tree = render
            .create(<AlertBar {...props} />)
            .toJSON();

        expect(tree).toMatchSnapshot();
    })
});

test('Action', async () => {
    const props = {
        visible: true,
        setVisible: jest.fn((bool) => visible = bool),
        severity: 'success',
        message: 'This is a test alert message'
    };

    const wrapper = mount(<AlertBar {...props} />);

    const closeButton = wrapper
        .find('Button')
        .filterWhere(button => button.text() === 'Close')
        .first();

    closeButton.props().onPress();
    wrapper.update();

    await new Promise(resolve => setImmediate(resolve));
    expect(visible).toBeFalsy();
})