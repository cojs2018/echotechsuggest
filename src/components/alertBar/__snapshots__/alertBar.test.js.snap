// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AlertBar renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#98fb98",
      "color": "#008000",
      "elevation": 1,
      "shadowColor": "#000000",
      "shadowOffset": Object {
        "height": 0.5,
        "width": 0,
      },
      "shadowOpacity": 0.24,
      "shadowRadius": 0.75,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "alignSelf": "center",
          "maxWidth": 960,
          "overflow": "hidden",
          "width": "100%",
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "height": 0,
        }
      }
    />
    <View
      onLayout={[Function]}
      style={Object {}}
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-start",
            "marginBottom": 0,
            "marginHorizontal": 8,
            "marginTop": 16,
          }
        }
      >
        <View
          style={
            Object {
              "margin": 8,
            }
          }
        >
          <Text
            accessibilityElementsHidden={true}
            allowFontScaling={false}
            importantForAccessibility="no-hide-descendants"
            pointerEvents="none"
            selectable={false}
            style={
              Array [
                Object {
                  "color": "#000000",
                  "fontSize": 40,
                },
                Array [
                  Object {
                    "lineHeight": 40,
                    "transform": Array [
                      Object {
                        "scaleX": 1,
                      },
                    ],
                  },
                  Object {
                    "backgroundColor": "transparent",
                  },
                ],
                Object {
                  "fontFamily": "Material Design Icons",
                  "fontStyle": "normal",
                  "fontWeight": "normal",
                },
                Object {},
              ]
            }
          >
            ó±‡­
          </Text>
        </View>
        <Text
          style={
            Array [
              Object {
                "color": "#000000",
                "fontFamily": "System",
                "fontWeight": "400",
                "textAlign": "left",
              },
              Object {
                "flex": 1,
                "margin": 8,
              },
            ]
          }
        >
          This is a test alert message
        </Text>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "margin": 4,
          }
        }
      >
        <View
          style={
            Object {
              "backgroundColor": "transparent",
              "borderColor": "transparent",
              "borderRadius": 4,
              "borderStyle": "solid",
              "borderWidth": 0,
              "elevation": 0,
              "margin": 4,
              "minWidth": "auto",
            }
          }
        >
          <View
            accessibilityRole="button"
            accessibilityState={
              Object {
                "disabled": undefined,
              }
            }
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                Object {
                  "borderRadius": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  undefined,
                ]
              }
            >
              <Text
                numberOfLines={1}
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#000000",
                      "fontFamily": "System",
                      "fontWeight": "400",
                      "textAlign": "left",
                    },
                    Array [
                      Object {
                        "letterSpacing": 1,
                        "marginHorizontal": 16,
                        "marginVertical": 9,
                        "textAlign": "center",
                      },
                      Object {
                        "marginHorizontal": 8,
                      },
                      Object {
                        "textTransform": "uppercase",
                      },
                      Object {
                        "color": "#6200ee",
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      Object {
                        "fontFamily": "System",
                        "fontWeight": "500",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                Close
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
